src/components/Transactions.tsx

```diff
@@ -0,0 +1,227 @@
+import { getEtherscanLink } from "../utils/getEtherscanLink"
+import { getPoolByAddress } from "../utils/getPoolByAddress"
+import { useActiveWeb3React } from "../hooks"
+import { useTranslation } from "react-i18next"
+
+interface Response {
+  data: {
+    addLiquidityEvents: [
+      {
+        swap: {
+          id: string
+        }
+        timestamp: string
+        transaction: string
+      },
+    ]
+    removeLiquidityEvents: [
+      {
+        swap: {
+          id: string
+        }
+        timestamp: string
+        transaction: string
+      },
+    ]
+    swaps: [
+      {
+        address: string
+        exchanges: [
+          {
+            boughtId: string
+            tokensBought: string
+            soldId: string
+            tokensSold: string
+            timestamp: string
+            transaction: string
+          },
+        ]
+      },
+    ]
+  }
+}
+
+interface Transaction {
+  object: string
+  transaction: string
+  time: string
+  timestamp: number
+  status?: string
+}
+
+export default function Transactions(): ReactElement {
+  const { t } = useTranslation()
+  const { chainId, account } = useActiveWeb3React()
+  const [transactionList, setTransactionList] = useState<Transaction[]>([])
+
+  function formatTime(timestamp: string) {
+    const timeoptions = {
+      month: "numeric",
+      day: "numeric",
+      hour: "2-digit",
+      minute: "2-digit",
+    } as const
+    return new Date(parseInt(timestamp) * 1000).toLocaleTimeString(
+      [],
+      timeoptions,
+    )
+  }
+
+  const fetchTxn = useCallback(async () => {
+    const url = "https://api.thegraph.com/subgraphs/name/saddle-finance/saddle"
+    const time24Hrs = Math.floor(Date.now() / 1000) - 60 * 60 * 24 // 24hrs
+    const query = account
+      ? `
+      {
+        addLiquidityEvents(where:{provider: "${account}", timestamp_gte: ${time24Hrs}}) {
+          swap {
+            id
+          }
+          transaction
+          timestamp
+        }
+        removeLiquidityEvents(where:{provider:"${account}", timestamp_gte: ${time24Hrs}}) {
+          swap {
+            id
+          }
+          transaction
+          timestamp
+        }
+        swaps {
+          address
+          exchanges(where:{buyer:"${account}", timestamp_gte: ${time24Hrs}}) {
+            boughtId
+            soldId
+            timestamp
+            transaction
+          }
+        }
+      }
+    `
+      : null
+
+    await fetch(url, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ query }),
+    })
+      .then((res) => res.json())
+      .then((result: Response) => {
+        const addLiquidityEvents = result.data.addLiquidityEvents
+        addLiquidityEvents.map((event) => {
+          const poolName = chainId
+            ? getPoolByAddress(event.swap.id, chainId)?.name
+            : null
+          poolName &&
+            setTransactionList((prevState) => [
```

Comment:
instead of calling `setTransactionList` to update state for every processed event (which will trigger many re-renders), it would be preferable to build the list in a variable and then call `setTransactionList` only once:
```js
const newTransactionList = []
newTransactionList.append({...})
setTransactionList(newTransactionList.sort(...))
```
this would also allow you to sort by timestamp here instead of in the JSX

Reply:
done